diff --git a/compiled/watchpack/index.js b/compiled/watchpack/index.js
index bef6fc7d6531244c0ffbcb6ed2fbae0c8461f2b7..1aab1c11975c53a28b59c1e55a0b9048204055d3 100644
--- a/compiled/watchpack/index.js
+++ b/compiled/watchpack/index.js
@@ -550,7 +550,10 @@ class DirectoryWatcher extends EventEmitter {
 						if (
 							err.code !== "ENOENT" &&
 							err.code !== "EPERM" &&
-							err.code !== "EBUSY"
+							err.code !== "EBUSY" &&
+							err.code !== "EACCES" &&
+							err.code !== "ENODEV" &&
+							!(err.code === "EINVAL" && process.platform === "win32")
 						) {
 							this.onStatsError(err);
 						} else {
@@ -777,7 +780,9 @@ class DirectoryWatcher extends EventEmitter {
 								err2.code === "ENOENT" ||
 								err2.code === "EPERM" ||
 								err2.code === "EACCES" ||
-								err2.code === "EBUSY"
+								err2.code === "EBUSY" ||
+								err2.code === "ENODEV" ||
+								(err2.code === "EINVAL" && process.platform === "win32")
 							) {
 								this.setMissing(itemPath, initial, "scan (" + err2.code + ")");
 							} else {
